tests/src/application/controllers/signup.spec.ts

```diff
@@ -1,295 +1,96 @@
-      const { sut, passwordValidatorStub } = makeSut()
-      jest.spyOn(passwordValidatorStub, 'isValid').mockReturnValueOnce(false)
-
-      const httpRequest = makeFakeSignUpRequest({ fields: { password: 'invalid_password', passwordConfirmation: 'invalid_password' } })
-
-      const httpResponse = await sut.handle(httpRequest)
-      expect(httpResponse).toEqual(badRequest(new InvalidParamError('password')))
-    })
-
-    it('Should call PasswordValidator with correct password', async () => {
-      const { sut, passwordValidatorStub } = makeSut()
-      const isValidSpy = jest.spyOn(passwordValidatorStub, 'isValid')
-
-      const httpRequest = makeFakeSignUpRequest({ fields: { password: 'valid_password', passwordConfirmation: 'valid_password' } })
-
-      await sut.handle(httpRequest)
-      expect(isValidSpy).toHaveBeenCalledWith('valid_password')
-    })
-
-    it('Should return 500 if PasswordValidator throws', async () => {
-      const { sut, passwordValidatorStub } = makeSut()
-      jest.spyOn(passwordValidatorStub, 'isValid').mockImplementationOnce(() => {
-        throw new Error()
-      })
-
-      const httpRequest = makeFakeSignUpRequest()
-
-      const httpResponse = await sut.handle(httpRequest)
-      expect(httpResponse).toEqual(makeFakeServerError())
-    })
-  })
-
-  describe('test the passwordConfirmation field', () => {
-    it('Should return 400 if no password confirmation is provided', async () => {
-      const { sut } = makeSut()
-
-      const httpRequest = makeFakeSignUpRequest({ fields: { passwordConfirmation: undefined } })
-      const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(badRequest(new MissingParamError('passwordConfirmation')))
-    })
-
-    it('Should return 400 if password confirmation fails', async () => {
-      const { sut } = makeSut()
-
-      const httpRequest = makeFakeSignUpRequest({ fields: { passwordConfirmation: 'invalid_password' } })
-      const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(badRequest(new InvalidParamError('passwordConfirmation')))
-    })
-  })
-
-  describe('test the isPrivacyPolicyAccepted field', () => {
-    it('Should return 400 if isPrivacyPolicyAccepted is not provided', async () => {
-      const { sut } = makeSut()
-
-      const httpRequest = makeFakeSignUpRequest({ fields: { isPrivacyPolicyAccepted: undefined } })
-      const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(badRequest(new MissingParamError('isPrivacyPolicyAccepted')))
-    })
-
-    it('Should return 400 if isPrivacyPolicyAccepted is false', async () => {
-      const { sut } = makeSut()
-
-      const httpRequest = makeFakeSignUpRequest({ fields: { isPrivacyPolicyAccepted: false } })
+  const httpRequest = makeFakeSignUpRequest()
+  describe('AddGuardian', () => {
+    it('Should return 403 (Conflict) if AddGuardian returns undefined', async () => {
```

Comment:
the HTTP status number for Conflict is 409 and not 403

Reply:
nice, Goondra, almost passes this one, I'll modify it here.